src/Native/Runtime/amd64/WriteBarriers.asm

```diff
@@ -425,6 +425,16 @@ LEAF_ENTRY RhpByRefAssignRef, _TEXT
     cmp     rcx, [g_ephemeral_high]
     jae     RhpByRefAssignRef_NotInHeap
 
+
+ifdef CORERT
+    ;; On CoreRT, JIT uses this helper to update stack locations.
+    ;; Check whether the writes were even into the heap. If not there's no card update required.
+    cmp     rdi, [G_LOWEST_ADDRESS]
```

Comment:
I would make this check before `g_ephemeral_low/high` so that the order matches how this helper is implemented in CoreCLR (https://github.com/dotnet/coreclr/blob/master/src/vm/amd64/jithelpers_fast.S#L148).